rust-runtime/aws-smithy-wasm/src/wasi.rs

```diff
@@ -0,0 +1,345 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! WASI HTTP Adapter
+use aws_smithy_http::header::ParseError;
+use aws_smithy_runtime_api::{
+    client::{
+        http::{
+            HttpClient, HttpConnector, HttpConnectorFuture, HttpConnectorSettings,
+            SharedHttpClient, SharedHttpConnector,
+        },
+        orchestrator::HttpRequest,
+        result::ConnectorError,
+        runtime_components::RuntimeComponents,
+    },
+    http::Response,
+    shared::IntoShared,
+};
+use aws_smithy_types::body::SdkBody;
+use bytes::{Bytes, BytesMut};
+use wasi::http::{
+    outgoing_handler,
+    types::{self as wasi_http, OutgoingBody, RequestOptions},
+};
+
+/// Builder for [`WasiHttpClient`]. Currently empty, but allows for future
+/// config options to be added in a backwards compatible manner.
+#[derive(Default, Debug)]
+pub struct WasiHttpClientBuilder {}
```

Comment:
So that it can't be constructed without calling `new` or `default`:
```suggestion
#[non_exhaustive]
pub struct WasiHttpClientBuilder {}
```